app.js

```diff
@@ -168,11 +168,22 @@ var eventsModule = function () {
     // used to return all of the EF templates used as array
     function GetTemplates()
     {
-        return Object.keys(efDataHolder).unshift("None");
+        templates = ["None"];
+        for(var t in efDataHolder){
+        templates.push(t);
+        }
+        return templates;     
```

Comment:
i noticed the previous code could cause problems too, looks like `unshift` returns the length of the newly created array. The following also would also work:

```js
templates = Object.keys(efDataHolder);
templates.unshift("None");
return templates
```